src/aggregation/contribution_tracker.c

```diff
@@ -147,14 +152,15 @@ ContributionTrackerState *contribution_tracker_new(
   return state;
 }
 
-void contribution_tracker_update_aid(ContributionTrackerState *state, aid_t aid)
+void contribution_tracker_update_aid(ContributionTrackerState *state, aid_t aid, contribution_t zero_contribution)
 {
   bool found;
   ContributionTrackerHashEntry *entry = ContributionTracker_insert(state->contribution_table, aid, &found);
   if (!found)
   {
     state->aid_seed ^= aid;
     entry->has_contribution = false;
+    entry->contributor.contribution = zero_contribution;
```

Comment:
this is the fix. I peeked into `simplehash.h` and, insofar as my C goes, `entry` can indeed be uninitialized memory.

Reply:
oh, not that it matters a big deal, but the bug might have actually been introduced just before, in #351, I only now realized this. But I'm only 80% sure of this - maybe it just didn't hit the assertion.

Reply:
The end result is a bit weird, since the entry is marked as having no contribution, but at the same time we set a contribution of zero. The `has_contribution` flag should act as a `NULL` indicator and the `contribution` field should be ignored entirely when the flag is not set.

If we don't want to differentiate between no and zero contributions anymore, maybe we should also drop the flag?
But that would prevent the correct handling of negative contributions (since a zero contribution is greater than all negative ones), so that doesn't seem right either.

Reply:
> but at the same time we set a contribution of zero.

This particular line is only responsible for "cleaning up" the init value inside `entry`. This makes me think, that the issue lies with a lack of `if (has_contribution)` check somewhere in the code

Reply:
OK, I've got one alternative fix, hold on a sec

Reply:
Turns out, I don't have an alternative fix.

The issue is with that `top_contributors` doesn't track `has_contribution` at all. Since we need the zero-contributions tracked in `top_contributors`, in case their AIDs need to be included for seeding (see other comment), we need to have a proper `contribution` field value for the sake of `top_contributors` not being polluted with garbage from the `simplehash.h` structure. 

Does this make sense?

OTOH, I don't see how this `top_contributors` could handle negative contributions? Should it collect and track absolute values of contributions?

Reply:
I was wrong, we don't need to handle negative values at all. For `sum`, we separate the negative values from the positive ones and we track them separately.
Then how about we get rid off the `has_contribution` flag and the `contribution_tracker_update_aid` function? They don't seem useful anymore. `contribution_tracker_update_contribution` will also need some refactoring.

Reply:
something like this? https://github.com/diffix/pg_diffix/pull/352/commits/27e2e0e93521cc9daa4301dbbbec44f136106f9d `has_contribution` seems to have served as a substitute for comparing `contribution_t` union against zero. So far I'm assuming that we do need the union, but maybe we do not and things will simplify further?

Reply:
And here is the follow-up bebe4aa9cba6db7c4a3422f1481a3c78d1bb5b1c. It's hard for me to judge which is more suitable, since it looks like this code is intended to handle cases somewhat beyond what it's currently used for. The index check added in this commit looks a bit mysterious, but it does the trick. WDYT?

Reply:
The union is needed because the contribution (in theory) can have different types (integer, real, etc.)